tests/utils.py

```diff
@@ -11,6 +15,17 @@ def lookup_sync_backend():
     return "sync"
 
 
+def wait_until_pproxy_serve_on_port(host: str, port: int):
```

Comment:
(Nit) Looks like there's nothing particular to pproxy here, so rather should we name this `_wait_can_connect(host, port)`, or something more generic like that? (With `_` indicating this is not a helper we're meaning to import from conftest or our tests.)

Reply:
Good point, renamed